src/order/locally_finite.lean

```diff
@@ -482,6 +482,239 @@ locally_finite_order.of_Icc' (α × β)
 
 end preorder
 
+/-!
+#### `with_top`, `with_bot`
+
+Adding a `⊤` to a locally finite `order_top` keeps it locally finite.
+Adding a `⊥` to a locally finite `order_bot` keeps it locally finite.
+-/
+
+section order_top
+open with_top
+variables (α) [order_top α] [locally_finite_order α]
+
+instance : locally_finite_order (with_top α) :=
+{ finset_Icc := λ a b, match a, b with
+    |      ⊤,      ⊤ := {⊤}
+    |      ⊤, some b := ∅
+    | some a,      ⊤ := cons (⊤ : with_top α) ((Ici a).map embedding.some)
+                              (λ h, let ⟨x, _, hx⟩ := mem_map.1 h in coe_ne_top hx)
+    | some a, some b := (Icc a b).map embedding.some
+    end,
```

Comment:
Turns out you can actually write `coe` instead of `some`:

```suggestion
local attribute [pattern] coe

instance : locally_finite_order (with_top α) :=
{ finset_Icc := λ a b, match a, b with
    |       ⊤,       ⊤ := {⊤}
    |       ⊤, (b : α) := ∅
    | (a : α),       ⊤ := cons (⊤ : with_top α) ((Ici a).map embedding.some)
                               (λ h, let ⟨x, _, hx⟩ := mem_map.1 h in coe_ne_top hx)
    | some a, some b := (Icc a b).map embedding.some
    end,
```